packages/axelar-wasm-std/src/utils.rs

```diff
@@ -19,6 +22,22 @@ impl<T> TryMapExt<T> for Vec<T> {
     }
 }
 
+/// since the `Felt` type doesn't error on overflow, we have to implement that check
+pub fn does_felt_overflow_from_slice(felt_hex_slice: &[u8]) -> bool {
+    if felt_hex_slice.len() > 32 {
+        return true;
+    }
```

Comment:
this case would be caught by the U256 comparison anyway right? I'd get rid of this

Reply:
`U256::from_be_slice` panics and I prefer to not handle an error or return a result and just keep it a simple bool.
Does that make sense?

PS: I'd put these checks into the custom type you suggest and this might be an unneeded discussion anyway.